File: dataset/avatar/Python/Code/codeforces_267_A.py
def substract ( a , b ) :
    if a > b :
        total = 0
        while a > 0 and b > 0 :
            total += int ( a / b )
            a = a % b
            if a > 0 :
                total += int ( b / a )
                b = b % a
        return total
    if a < b :
        total = 0
        while a > 0 and b > 0 :
            total += int ( b / a )
            b = b % a
            if b > 0 :
                total += int ( a / b )
                a = a % b
        return total
    if a == b :
        return 1
n = int ( input ( ) )
lst = [ ]
for i in range ( 0 , n ) :
    a , b = map ( int , input ( ).split ( ) )
    lst.append ( substract ( a , b ) )
for i in lst :
    print ( i )

File: output/gpt_4o/avatar/direct_translation/python/java/codeforces_267_A.java

import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class codeforces_267_A {
    public static int substract(int a, int b) {
        if (a > b) {
            int total = 0;
            while (a > 0 && b > 0) {
                total += a / b;
                a = a % b;
                if (a > 0) {
                    total += b / a;
                    b = b % a;
                }
            }
            return total;
        }
        if (a < b) {
            int total = 0;
            while (a > 0 && b > 0) {
                total += b / a;
                b = b % a;
                if (b > 0) {
                    total += a / b;
                    a = a % b;
                }
            }
            return total;
        }
        if (a == b) {
            return 1;
        }
        return 0;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = Integer.parseInt(scanner.nextLine());
        List<Integer> lst = new ArrayList<>();
        for (int i = 0; i < n; i++) {
            String[] input = scanner.nextLine().split(" ");
            int a = Integer.parseInt(input[0]);
            int b = Integer.parseInt(input[1]);
            lst.add(substract(a, b));
        }
        for (int result : lst) {
            System.out.println(result);
        }
        scanner.close();
    }
}

